code/game/mecha/mecha_decal_kits.dm

```diff
@@ -0,0 +1,561 @@
+                    if(do_after_once(usr, 15, target = current_mecha))
+                        current_mecha.glow_colour = chosencolour
+                        colour_overlay(current_mecha.glow_colour)
+                        to_chat(usr, "<span class='notice'>You carefully cover the exosuit in a new coat of paint.</span>")
+                        playsound(loc, spray_sound, 50, 1, -6)
+                        openUI(current_mecha, usr)
+            if("default_base")
+                if(current_mecha.Adjacent(usr) && do_after_once(usr, 15, target = current_mecha))
+                    current_mecha.basecoat_colour = initial(current_mecha.basecoat_colour)
+                    to_chat(usr, "<span class='notice'>You carefully cover the exosuit in a new coat of paint.</span>")
+                    playsound(loc, spray_sound, 50, 1, -6)
+                    openUI(current_mecha, usr)
+            if("default_glow")
+                if(current_mecha.Adjacent(usr) && do_after_once(usr, 15, target = current_mecha))
+                    current_mecha.glow_colour = initial(current_mecha.glow_colour)
+                    to_chat(usr, "<span class='notice'>You carefully cover the exosuit in a new coat of paint.</span>")
+                    playsound(loc, spray_sound, 50, 1, -6)
+                    openUI(current_mecha, usr)
+            if("defaultdecals")
+                if(current_mecha.Adjacent(usr) && do_after_once(usr, 15, target = current_mecha))
+                    current_mecha.decals = list()
+                    for(var/datum/mecha/mecha_decal/MD in current_mecha.default_decals)
+                        current_mecha.decals.Add(MD.clone())
+                    current_mecha.basecoat_colour = initial(current_mecha.basecoat_colour)
+                    current_mecha.glow_colour = initial(current_mecha.glow_colour)
+                    to_chat(usr, "<span class='notice'>You strip all the decals off [current_mecha] and return it to its original colour scheme.</span>")
+                    playsound(loc, spray_sound, 50, 1, -6)
+                    openUI(current_mecha, usr)
+            if("decalsave")
+                if(current_mecha.Adjacent(usr))
+                    var/input_text = copytext(reject_bad_text(input(usr,"Pattern name?","Set Pattern Name","")),1,MAX_NAME_LEN)
+                    if(input_text)
+                        save_pattern(current_mecha, current_mecha.decals, input_text) 
+                        openUI(current_mecha, usr)
+        current_mecha.redraw_cache()
+    
+    if(href_list["decalcolour"])
+        if(current_mecha.occupant)
+            to_chat(usr, "<span class='warning'>The exosuit must be unoccupied before you can paint it!</span>")
+            return
+        if(current_mecha.Adjacent(usr))
+            var/hrefstring = href_list["decalcolour"]
+            var/datum/mecha/mecha_decal/D = locate(hrefstring)
```

Comment:
Will have to check if D is still there. It is possible that it no longer exists once this code hits and that will cause runtimes. Same applies to the other locates